Dockerfile.windows

```diff
@@ -1,57 +1,96 @@
+# balk at reparse points. Again, see the example at the top of this file on how use a volume
+# to get the built binary out of the container.
 
 FROM windowsservercore
 
-ENV GO_VERSION=1.5.3 \
+# Environment variable notes:
+#  - GOLANG_VER should be updated to be consistent with the Linux dockerfile.
+#  - FROM_DOCKERFILE is used for detection of building within a container.
+ENV GOLANG_VERSION=1.5.3 \
     GIT_VERSION=2.7.0 \
-    RSRC_COMMIT=e48dbf1b7fc464a9e85fcec450dddf80816b76e0 \
+    RSRC_COMMIT=ba14da1f827188454a4591717fff29999010887f \
     GOPATH=C:/go;C:/go/src/github.com/docker/docker/vendor \
     FROM_DOCKERFILE=1
 
+# Make sure we're in temp for the downloads
 WORKDIR c:/windows/temp
 
-RUN powershell -command \
-    sleep 2 ; \
-    iex ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1'))
-
-RUN powershell -command \
-    sleep 2 ; \
-    choco install curl -y -v ; \
-    choco install git -y -v --version=%GIT_VERSION% -params '"/GitAndUnixToolsOnPath"'
-
-RUN setx /M Path "c:\ProgramData\chocolatey\bin;c:\Program Files\Git\usr\bin;%Path%;c:\gcc\bin"
+# Chocolatey and install curl/git
+RUN powershell -command sleep 5 ; iex ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1'))
+RUN powershell -command sleep 5; choco install curl -y -v
+RUN powershell -command sleep 5; choco install git -y -v --version=%GIT_VERSION% -params '"/GitAndUnixToolsOnPath"'
 
-RUN powershell -command \
-    sleep 2; \
-    curl.exe -L -k -o go.msi https://storage.googleapis.com/golang/go%GO_VERSION%.windows-amd64.msi; \
-    curl.exe -L -k -o gcc.zip http://downloads.sourceforge.net/project/tdm-gcc/TDM-GCC%205%20series/5.1.0-tdm64-1/gcc-5.1.0-tdm64-1-core.zip ; \
-    curl.exe -L -k -o runtime.zip http://downloads.sourceforge.net/project/tdm-gcc/MinGW-w64%20runtime/GCC%205%20series/mingw64runtime-v4-git20150618-gcc5-tdm64-1.zip ; \
-    curl.exe -L -k -o binutils.zip http://downloads.sourceforge.net/project/tdm-gcc/GNU%20binutils/binutils-2.25-tdm64-1.zip
+# Path
+RUN setx /M Path "c:\ProgramData\chocolatey\bin;c:\Program Files\Git\usr\bin;%Path%;c:\gcc\bin;c:\7zip"
 
-RUN powershell -command \
-    .\go.msi /quiet ; \
-    Expand-Archive gcc.zip \gcc -Force ; \
-    Expand-Archive runtime.zip \gcc -Force ; \
-    Expand-Archive binutils.zip \gcc -Force
+# Download everything else we need to install
+# We want a 64-bit make.exe, not 16 or 32-bit. This was hard to find, so documenting the links
+#  - http://sourceforge.net/p/mingw-w64/wiki2/Make/ -->
+#  - http://sourceforge.net/projects/mingw-w64/files/External%20binary%20packages%20%28Win64%20hosted%29/ -->
+#  - http://sourceforge.net/projects/mingw-w64/files/External binary packages %28Win64 hosted%29/make/
+RUN powershell -command sleep 5; curl.exe -L -k -o make.zip http://downloads.sourceforge.net/project/mingw-w64/External%20binary%20packages%20%28Win64%20hosted%29/make/make-3.82.90-20111115.zip
```

Comment:
It's a Windows port. Sure, I'll investigate -fsSL (need to look up what that does though and if supported by this version of curl)